### ISSUE TITLE
Add disclose indicator

### ISSUE THREAD

**Issue by cukingpro** [pitiphong-p/CollectionViewShelfLayout#12]
Hi,
Can we add disclose indicator for a section like tableView cell?

**Comment by pitiphong-p** [comment]
Yes,  you can. By having a custom subclass for the Header view for each section

**Comment by cukingpro** [comment]
But the header is on top. Can I put it to the right like this?
<img width="597" alt="Screen Shot 2019-08-07 at 3 12 09 PM" src="https://user-images.githubusercontent.com/9702625/62606280-185aa400-b926-11e9-9ed1-2cd2acec42b4.png">

**Comment by pitiphong-p** [comment]
What are those square? Are they a UICollectionViewCell?

**Comment by cukingpro** [comment]
Yes. Those are cells of that section.

**Comment by pitiphong-p** [comment]
I think in this case, you can have a disclosure indicator a UICollectionViewCell with another cell type with its own cell identifier and return it from the cellForItem method.

**Comment by cukingpro** [comment]
That will make it scrollable. I want it to stay at the right.

**Comment by cukingpro** [comment]
<img width="975" alt="Screen Shot 2019-08-07 at 3 28 12 PM" src="https://user-images.githubusercontent.com/9702625/62607307-0da10e80-b928-11e9-805f-2f239b0eff56.png">
This is the design.

**Comment by pitiphong-p** [comment]
I see. I'm afraid that this library doesn't support this scenario out of the box but you can fork this project and make a change to make it support your scenario

**Comment by cukingpro** [comment]
Thank you for your work. Do you have any suggestion for that?

**Comment by pitiphong-p** [comment]
You can add a new kind of supplementary view for the detail disclosure (and may also with a custom Layout Attribute) and layout it in prepareLayout method similar to https://github.com/pitiphong-p/CollectionViewShelfLayout/blob/master/CollectionViewShelfLayout/CollectionViewShelfLayout.swift#L178-L186 and also return in the layoutAttributesForElements(in:) and layoutAttributesForSupplementaryView(ofKind:at:) methods

**Comment by pitiphong-p** [comment]
Or if you set the deployment target at iOS 13 then you may try the new https://developer.apple.com/documentation/uikit/uicollectionviewcompositionallayout as presented in https://developer.apple.com/videos/play/wwdc2019/215/
